scene/gui/file_dialog.cpp

```diff
@@ -356,6 +356,8 @@ void FileDialog::_action_pressed() {
 	if (mode == FILE_MODE_SAVE_FILE) {
 		bool valid = false;
 
+		bool use_case_sensitive_filter = case_sensitive_filter == CASE_SENSITIVITY_TRUE || case_sensitive_filter == CASE_SENSITIVITY_OS_DEFAULT && dir_access->is_case_sensitive(f);
```

Comment:
```suggestion
		bool use_case_sensitive_filter = case_sensitive_filter == CASE_SENSITIVITY_TRUE || (case_sensitive_filter == CASE_SENSITIVITY_OS_DEFAULT && dir_access->is_case_sensitive(f));
```
Or
```suggestion
		bool use_case_sensitive_filter = (case_sensitive_filter == CASE_SENSITIVITY_TRUE || case_sensitive_filter == CASE_SENSITIVITY_OS_DEFAULT) && dir_access->is_case_sensitive(f);
```
Depending on what you indented, it's ambiguous